include/boost/geometry/strategy/cartesian/side_by_triangle.hpp

```diff
@@ -205,23 +206,27 @@ public :
     template <typename P1, typename P2, typename P>
     static inline int apply(P1 const& p1, P2 const& p2, P const& p)
     {
-        using coor_t = typename select_calculation_type_alt<CalculationType, P1, P2, P>::type;
-
-        // Promote float->double, small int->int
-        using promoted_t = typename select_most_precise<coor_t, double>::type;
-
-        bool const are_all_integral_coordinates =
+        constexpr bool are_all_integral_coordinates =
             std::is_integral<coordinate_type_t<P1>>::value
             && std::is_integral<coordinate_type_t<P2>>::value
             && std::is_integral<coordinate_type_t<P>>::value;
 
+        // Promote float->double, integers to std::int64_t
```

Comment:
Is the latter part of this comment true/intended to remain true, given 
```
// For now deactivate the use of multiprecision integers
// TODO: activate it later
#define BOOST_GEOMETRY_NO_MULTIPRECISION_INTEGER
```
<https://github.com/boostorg/geometry/blob/0900b5d61f6c0b304e78475d8e56a34cf168170d/include/boost/geometry/util/promote_integral.hpp#L14>
 for the case in #629 where std::int64_t is used ? Or is that TODO obsolete?

Reply:
Thanks for your sharp observation.
I will change `int64_t` in the comments (this was my first approach) because integer will go to 'long'.
It is actually not related to `BOOST_GEOMETRY_NO_MULTIPRECISION_INTEGER` because that is really about the Boost.Multi Precision library and their integers. That is turned off indeed (I don't remember the `TODO` thanks - I think we should not turn it on automatically, it should be a conscious choice for users if they want it, or not).

So what happens now (at least on my mac) is:
`short` will to to `int`
`int` will go to `long` (instead of `double` before - even though the comment said it would not)
`int64` will stay `int64_t` (instead of `double` before)

Reply:
The full list (I added code to display it in a comment in #629)
```
t: s -> promoted_t: i
t: i -> promoted_t: l
t: l -> promoted_t: l
t: x -> promoted_t: x
t: x -> promoted_t: x
```

Reply:
I see, thanks. My concern was, if it was a TODO to remove that macro, if it were removed then int64_t would actually be promoted to a multiprecision guide because it is in the promotion list if that macro is not defined https://github.com/boostorg/geometry/blob/0900b5d61f6c0b304e78475d8e56a34cf168170d/include/boost/geometry/util/promote_integral.hpp#L237 . I agree that it should not be turned on by default.

Reply:
Agreed, I changed the todo now in this PR. I didn't want to make the real change now but created another ticket #1380 to not forget it.